### PR TITLE
Adds tests for NewsletterFooter and NewsletterForm (closes #1640).

### LINKED ISSUE #1640: Write tests for newsletter form in the footer.
Followup from https://github.com/mozilla/testpilot/pull/1620

### PR DESCRIPTION
**Pull request by chuckharmston**

### CODE REVIEW

File: frontend/src/app/actions/newsletter-form.js [hunk 1]
```diff
@@ -1,4 +1,7 @@
 import { createActions } from 'redux-actions';
+import 'isomorphic-fetch';
```
Comment (chuckharmston) [inline frontend/src/app/actions/newsletter-form.js:2]: Even though this call defines a global in the bundled module, it's necessary for this to be imported in the module for the mock to work. It shouldn't affect the bundle.

File: frontend/test/app/actions/newsletter-form-test.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+const SUBMITTING = 'newsletterFormSetSubmitting';
+const SUCCEEDED = 'newsletterFormSetSucceeded';
+
+const MOCK_EMAIL = 'foo@bar.com';
+
+const setUp = responseCode => {
+  fetchMock.post(basketUrl, responseCode);
+  return {
+    dispatch: sinon.spy(),
+    subscribe: require('../../../src/app/actions/newsletter-form').default.newsletterFormSubscribe
```
Comment (chuckharmston) [inline frontend/test/app/actions/newsletter-form-test.js:18]: Has to be imported after the `fetchMock.post` call.

A little ugly, but everything else mocks _so cleanly_ that it's probably fine.

**Comment by chuckharmston** [comment]
I'm really happy with how this turned out; the tests are clean, clear, and fast. Great job setting up the harness @lmorchard.

**Review by lmorchard — APPROVED** [review]
Still a bit meh about the the work to mock out fetch() vs the higher-level subscribe(), but this works & catches all the things so :+1:

**Pull request merged**